src/_pytest/python.py

```diff
@@ -545,15 +544,22 @@ def _importtestmodule(self):
 
 
 class Package(Module):
-    def __init__(self, fspath, parent=None, config=None, session=None, nodeid=None):
+    def __init__(
+        self,
+        fspath: py.path.local,
+        parent: nodes.Node,
```

Comment:
:+1:  on removing the default, seems like it didn't make sense given it's dereferenced just below.

BTW, as far I managed to gather, `nodes.Node` is not a "real" type, but only meant to be a common subclass for `nodes.Item` and `nodes.Collector` (which are themselves only meant to be subclasses). And `nodes.Item` is meant for leaf nodes, and `nodes.Collector` for non-leaf nodes. If that's true, then it doesn't make sense for `parent` to be a `nodes.Item`, only `nodes.Collector`, so can switch to that instead of `nodes.Node`.

Reply:
Makes sense, amended.